lambdas/archive-vcs/src/main/java/uk/gov/di/ipv/core/archivevcs/ArchiveVcsHandler.java

```diff
@@ -71,32 +87,80 @@ public void handleRequest(InputStream inputStream, OutputStream outputStream, Co
+    private void revoke(String userId) throws Exception {
         // Read KBV VC for the ID
-        VcStoreItem vc = this.verifiableCredentialService.getVcStoreItem(userId, "kbv");
-        if (vc != null) {
+        VcStoreItem vcStoreItem = this.verifiableCredentialService.getVcStoreItem(userId, "kbv");
+
+        if (vcStoreItem != null) {
             // Archive VC
-            dataStore.create(vc);
+            dataStore.create(vcStoreItem);
 
             // Submit audit event
-            // ...
+            sendVcRevokedAuditEvent(userId, vcStoreItem);
 
             // Delete VC from the main table
             // ...
 
         } else {
-            LOGGER.error("VC {} cannot be found", userId);
+            throw new Exception("VcStoreItem cannot be found");
+        }
+    }
+
+    private void sendVcRevokedAuditEvent(String userId, VcStoreItem vcStoreItem) throws ParseException, SqsException, JsonProcessingException {
+        var auditEventUser = new AuditEventUser(userId, null, null, null);
+
+        var signedCredential = SignedJWT.parse(vcStoreItem.getCredential());
+        var jwtClaimsSet = signedCredential.getJWTClaimsSet();
+        var vc = (JSONObject) jwtClaimsSet.getClaim("vc");
+        var evidence = vc.getAsString("evidence");
+
+        var auditExtensions = new AuditExtensionsVcEvidence(jwtClaimsSet.getIssuer(), evidence);
+
+        auditService.sendAuditEvent(
+                new AuditEvent(
+                        AuditEventTypes.IPV_VC_REVOKED,
+                        configService.getSsmParameter(ConfigurationVariable.COMPONENT_ID),
+                        auditEventUser,
+                        auditExtensions));
+    }
+
+    private void sendRevokedFailureAuditEvent(String userId, String errorMessage, int i, int userIdsSize) {
+        var auditEventUser = new AuditEventUser(userId, null, null, null);
+
+        var extensionErrorParams = new AuditExtensionErrorMessage(
+                errorMessage
+        );
+
+        var auditEvent =
+                new AuditEvent(
+                        AuditEventTypes.IPV_VC_REVOKED_FAILURE,
+                        configService.getSsmParameter(ConfigurationVariable.COMPONENT_ID),
+                        auditEventUser,
+                        extensionErrorParams);
+        LOGGER.info(
+                LogHelper.buildLogMessage(
+                        String.format(
+                                "Sending audit event IPV_VC_REVOKED_FAILURE message at index %s / %s.",
+                                i, userIdsSize)));
+        try {
+            auditService.sendAuditEvent(auditEvent);
+        } catch (SqsException sqsException) {
+            LOGGER.error(
+                    LogHelper.buildLogMessage(
+                            String.format(
+                                    "Failed to send audit event at index %s / %s.",
+                                    i, userIdsSize)));
+            throw new RuntimeException();
```

Comment:
In the case that sending audit events doesn't work, we will log the progress through the list (no userId but it can be worked out from the input list), then throw an unhandled exception to stop this issue from continuing to happen